src/Analysis/Codelyzer.Analysis.Build/WorkspaceBuilderHelper.cs

```diff
@@ -433,6 +442,49 @@ private void BuildSolution(IAnalyzerManager manager)
             }
         }
 
+        private void BuildSolutionOnlyWithoutOutput(string solutionPath, bool isFramework)
+        {
+            try
+            {
+                System.Diagnostics.Process process = new System.Diagnostics.Process();
+                System.Diagnostics.ProcessStartInfo startInfo = new System.Diagnostics.ProcessStartInfo();
+                startInfo.WindowStyle = System.Diagnostics.ProcessWindowStyle.Hidden;
+
+                var msBuildLocation = _analyzerConfiguration.BuildSettings.MSBuildPath;
+                if (string.IsNullOrEmpty(msBuildLocation))
+                {
+                    msBuildLocation = _msBuildDetector.GetFirstMatchingMsBuildFromPath();
+                }
+
+                //Add quotes around solution name to make sure spaces don't cause the command to fail
+                var escapedSolutionPath = "\"" + solutionPath + "\"";
+
+                var arguments = new List<string>(_analyzerConfiguration.BuildSettings.BuildArguments);
+
+                if (isFramework)
+                {
+                    startInfo.FileName = msBuildLocation;
```

Comment:
what if the msBuildLocation is empty ?

Reply:
If an msbuild location is not specified, we go back to retrieve it from the file system (check line #456 above). If both didn't retrieve a value, then this is an exception and we're expecting this to fail. Catching at this level for this API will hide the error, because we're not returning anything.

Reply:
We should throw exception. Ill add as a new PR